server/handlers.go

```diff
@@ -998,6 +1023,84 @@ func (s *Server) writeAccessToken(w http.ResponseWriter, idToken, accessToken, r
 	w.Write(data)
 }
 
+// Handler for the /userinfo endpoint
+func (s *Server) handleUserInfo(w http.ResponseWriter, r *http.Request) {
+
+	token := ""
+	if r.Header.Get("Authorization") != "" {
+		// Extract token type and token value from Authorization header
+		authorizationSubstrs := strings.Split(r.Header.Get("Authorization"), " ")
+		token = authorizationSubstrs[1]
+	} else if r.URL.Query().Get("access_token") != "" {
+		token = r.URL.Query().Get("access_token")
+	} else {
+		s.logger.Error("No token found")
+		return
+	}
+
+	decryptedToken, err := s.Decrypt(token)
+	if err != nil {
+		s.logger.Errorf("Internal server error: %v", err)
+		s.renderError(w, http.StatusInternalServerError, "Internal server error.")
+		return
+	}
+
+	accessToken := new(internal.AccessToken)
```

Comment:
I rarely see `new` used in Go, any reason not to just 
```suggestion
	accessToken := &internal.AccessToken{}
```

As we do with most other bare initialisations of structs?